src/rpc/test/chainStorageWatcher.test.ts

```diff
@@ -0,0 +1,235 @@
+import { makeAgoricChainStorageWatcher } from '../src/chainStorageWatcher';
+import { expect, it, describe, beforeEach, vi, afterEach } from 'vitest';
+import { AgoricChainStoragePathKind } from '../src/types';
+
+const fetch = vi.fn();
+global.fetch = fetch;
+global.harden = val => val;
+
+const fakeRpcAddr = 'https://agoric-rpc.vitest-nodes.com:443';
+const serialize = (val: unknown) => val;
+const unserialize = (val: any) => val;
+
+let watcher: ReturnType<typeof makeAgoricChainStorageWatcher>;
+
+describe('makeAgoricChainStorageWatcher', () => {
+  beforeEach(() => {
+    watcher = makeAgoricChainStorageWatcher(fakeRpcAddr, unserialize);
+    vi.useFakeTimers();
```

Comment:
does this need to be before *each* or could it be before all? everything in this file uses fake timers. does vitest leak that option to other tests? (iow what would happen if `vi.useRealTimers()` were omitted?)

Reply:
I'm not sure, does it matter? I'm just following the docs https://vitest.dev/guide/mocking.html#dates

Reply:
It matters to understanding Vitest execution model. I'm fine to cargo cult the docs for now.